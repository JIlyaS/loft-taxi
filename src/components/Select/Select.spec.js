import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Brightness1Icon from '@material-ui/icons/Brightness1';

import { Select } from './';

const defaultProps = {
  id: 'from',
  className: '',
  list: [],
  value: 'Пулково',
  iconStart: () => (
    <Brightness1Icon className="map-form__brightness-icon" />
  ),
  onChange: () => {}
};

const TestSelect = props => <Select {...defaultProps} {...props} />;

describe('Select', () => {
  it('renders correctly snapshot', () => {
      const tree = renderer.create(<TestSelect />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it("render card correctly with null props", () => {
      const props = {
        id: 'from',
        className: '',
        list: [],
        value: 'Пулково',
        iconStart: () => (
          <Brightness1Icon className="map-form__brightness-icon" />
        ),
        onChange: () => {}
      };
      const InputComponent = mount(<TestSelect {...props} />);
      expect(InputComponent.prop('id')).toEqual('from');
      expect(InputComponent.prop('value')).toEqual('Пулково');
    });
});
