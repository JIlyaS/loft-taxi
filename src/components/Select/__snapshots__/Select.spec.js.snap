// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly snapshot 1`] = `
<div
  className="makeStyles-root-1 select"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root map-form__brightness-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
        </svg>
      </div>
      <div
        aria-haspopup="listbox"
        aria-labelledby="from"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
        id="from"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        aria-hidden={true}
        autoFocus={false}
        className="MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
        value="Пулково"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
